src/main/kotlin/io/github/mojira/arisa/JiraConnectionService.kt

```diff
@@ -0,0 +1,134 @@
+package io.github.mojira.arisa
+
+import com.uchuhimo.konf.Config
+import io.github.mojira.arisa.infrastructure.config.Arisa
+import io.github.mojira.arisa.infrastructure.jira.connectToJira
+import java.lang.Long.max
+import java.time.Duration
+import java.time.Instant
+import java.util.concurrent.TimeUnit
+
+class JiraConnectionService(
+    private val config: Config
+) {
+    companion object {
+        /**
+         * How long ago the last successful connection needs to be for Arisa to try to relog
+         */
+        private const val MAX_SECONDS_SINCE_LAST_SUCCESSFUL_CONNECTION = 60L
+
+        /**
+         * How long Arisa should wait after it was unable to relog
+         */
+        private const val WAIT_TIME_AFTER_UNSUCCESSFUL_RELOG_IN_SECONDS = 5 * 60L // 5 minutes
+
+        /**
+         * How long Arisa should wait after a connection error before continuing without trying to relog
+         */
+        private const val WAIT_TIME_AFTER_CONNECTION_ERROR_IN_SECONDS = 40L
+    }
+
+    private var lastSuccessfulConnection = Instant.now().minusSeconds(MAX_SECONDS_SINCE_LAST_SUCCESSFUL_CONNECTION + 1)
```

Comment:
What is the rationale for this value? Relog immediately if initial connection attempt fails?
Might be good to add a comment explaining the value.

Reply:
Hmm, I'm not too sure myself. I think it's a remnant of debugging stuff, I was running into infinite loops when starting the bot and connection failed and this solved it. But I think the actual bug was elsewhere.

Reply:
Simply using `Instant.now()` seems to work just fine, not sure why I did it this way